### PR TITLE
Send MSP via backpack for DVR control

### PR DESCRIPTION
**Pull request by wvarty**
This PR adds 3x new parameters to LUA, under a new "Backpack" folder.

The 3x params relate the newly introduced (yet to be merged) backpack target for HDZero, which has an MSP message that allows for DVR recording to be started and stopped, with an optional delay (in seconds).
`DVR AUX` controls with channel will be used to start / stop the DVR recording. The switch point is at 1500uS, where >1500 is the "start" command, and <=1500 is the "stop" command. The AUX state can also be inverted by selecting the `!AUX` option for that channel.

`DVR Srt Dly` and `DVR Stp Dly` is the time in seconds that the command will be delayed on the VRX module itself.

The new code takes advantage of a callback in the CRSF lib that gets used by the BLE joystick when active. Since the DVR functionality is not expected to work while a user is using the BLE joystick, I have re-used this callback to monitor the CRSF channels, and respond if the AUX changes state.

Configurator PR for HDZ targets:
https://github.com/ExpressLRS/ExpressLRS-Configurator/pull/279

Backpack PR for HDZ targets:
https://github.com/ExpressLRS/Backpack/pull/60

### CODE REVIEW

File: src/lib/Backpack/devBackpack.cpp [hunk 1]
```diff
@@ -99,7 +103,81 @@ void BackpackBinding()
...
+        // DVR AUX control is off
+        return;
+    }
+
+    uint8_t auxNumber = (config.GetDvrAux() - 1) / 2 + 4;
+    uint8_t auxInverted = (config.GetDvrAux() + 1) % 2;
+
+    bool recordingState = CRSF_to_BIT(CRSF::ChannelDataIn[auxNumber]) ^ auxInverted;
+
+    if (recordingState == lastRecordingState)
```
[the remark this answers is not in the document]
Comment (wvarty) [reply src/lib/Backpack/devBackpack.cpp:152]: That means we will spam out this MSP message 500 times a second on 500Hz?
Seems hugely unnecessary.
Better off adding an ACK message to our backpack-to-backpack comms if you're worried about that?

[the remark this answers is not in the document]
Comment (JyeSmith) [reply src/lib/Backpack/devBackpack.cpp:152]: Ah, soz I miss understood the var.  This is just tracking switch position.

... carry on.

**Review by JyeSmith — APPROVED** [review]

File: src/lib/Backpack/devBackpack.cpp [hunk 2]
```diff
@@ -99,7 +103,81 @@ void BackpackBinding()
...
+    case 5:
+        return 60;
+        break;
+    case 6:
+        return 120;
+        break;
+    default:
+        return 0;
+        break;
+    }
```
Comment (pkendall64) [inline src/lib/Backpack/devBackpack.cpp:135]:
```cpp
constexpr uint8_t delays[] = {0, 5, 15, 30, 45, 60, 120};
return delays[index >= sizeof(delays) ? 0 : index];
```

Comment (wvarty) [reply src/lib/Backpack/devBackpack.cpp]: Changed

**Review by pkendall64 — APPROVED** [review]
Were you going to rebase this to a new branch for a 2.3 release or going to do a 2.3 release another way?

**Comment by wvarty** [comment]
> Were you going to rebase this to a new branch for a 2.3 release or going to do a 2.3 release another way?

I'm not wedded to any particularly approach, so whatever works best really...
Might be trivialising it, but I'd have thought the easiest way would be create a `2.3.x-maintenance` branch off `2.2.x-maintenance`, and cherry pick this over.

**Comment by pkendall64** [comment]
> > Were you going to rebase this to a new branch for a 2.3 release or going to do a 2.3 release another way?
> 
> I'm not wedded to any particularly approach, so whatever works best really... Might be trivialising it, but I'd have thought the easiest way would be create a `2.3.x-maintenance` branch off `2.2.x-maintenance`, and cherry pick this over.

That's fine. I just remember the conversation and you wanting a release for HDzero.

**Comment by wvarty** [comment]
> > > Were you going to rebase this to a new branch for a 2.3 release or going to do a 2.3 release another way?
> > 
> > 
> > I'm not wedded to any particularly approach, so whatever works best really... Might be trivialising it, but I'd have thought the easiest way would be create a `2.3.x-maintenance` branch off `2.2.x-maintenance`, and cherry pick this over.
> 
> That's fine. I just remember the conversation and you wanting a release for HDzero.

I guess the other option could be to rebase this PR on a branch off 2.2.x now, and them merge that up into master, in which case this PR would need recreating...
I'm happy with whatever...

**Comment by wvarty** [comment]
Merged to `2.3.x-maintenance` instead because (and I'm quoting) Jye was "being a little bitch" about this affecting the VTX+RX PR. Closing this PR since it has been approved, and is now in 2.3.
This will make it's way to `master` after we get Jye's PR in, at which point I'll deal with any merge conflicts, and add the LUA arrows.

**Pull request closed without merging**